model/model.go

```diff
@@ -98,14 +99,34 @@ type Container struct {
 	Ports        []ExposedPort `json:"ports" yaml:"ports"`
 	MemoryLimit  string        `json:"memoryLimit,omitempty" yaml:"memoryLimit,omitempty"`
 	MountSources bool          `json:"mountSources" yaml:"mountSources"`
+
+	// Base root command inside container
+	Command []string `json:"command" yaml:"command"`
+	// Arguments of the base root command inside container
+	Args []string `json:"args" yaml:"args"`
 }
 
 type ChePlugin struct {
-	ID           string      `json:"id" yaml:"id"`
-	Version      string      `json:"version" yaml:"version"`
-	Name         string      `json:"name" yaml:"name"`
-	Publisher    string      `json:"publisher" yaml:"publisher"`
-	Endpoints    []Endpoint  `json:"endpoints" yaml:"endpoints"`
-	Containers   []Container `json:"containers" yaml:"containers"`
-	WorkspaceEnv []EnvVar    `json:"workspaceEnv" yaml:"workspaceEnv"`
+	ID            string        `json:"id" yaml:"id"`
+	Version       string        `json:"version" yaml:"version"`
+	Name          string        `json:"name" yaml:"name"`
+	Type          string        `json:"type" yaml:"type"`
+	Publisher     string        `json:"publisher" yaml:"publisher"`
+	Endpoints     []Endpoint    `json:"endpoints" yaml:"endpoints"`
+	Containers    []Container   `json:"containers" yaml:"containers"`
+	WorkspaceEnv  []EnvVar      `json:"workspaceEnv" yaml:"workspaceEnv"`
+	PluginPatcher PluginPatcher `json:"pluginPatcher" yaml:"pluginPatcher"`
+}
+
+type PluginPatcher struct {
```

Comment:
Is the whole `PluginPatcher` structure really required to manage only the Che Theia use-case ?
Wasn't it simpler to just implement the change on the plugin-broker side for the Theia/Vscode plugins (since the broken already makes the difference between the plugin types) ?

Is the `PluginPatcher` type (that now leaks into the plugin registry `meta.yaml` json schema) sufficiently general to be really useful for other type of Che plugins ? I'm especially thinking of the `matcher`: maybe matching on the type only is a bit limited if you want the plugin patcher to be a general and generic mechanism...